tests/test-sessions/test-infinispan-sessions/src/test/java/org/eclipse/jetty/server/session/remote/RemoteInfinispanTestSupport.java

```diff
@@ -31,42 +31,93 @@
 import org.hibernate.search.cfg.SearchMapping;
 import org.infinispan.client.hotrod.RemoteCache;
 import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ClientIntelligence;
+import org.infinispan.client.hotrod.configuration.Configuration;
 import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
 import org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller;
 import org.infinispan.protostream.FileDescriptorSource;
 import org.infinispan.protostream.SerializationContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.MountableFile;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.junit.jupiter.api.Assertions.fail;
 
 /**
  * RemoteInfinispanTestSupport
  */
 public class RemoteInfinispanTestSupport
 {
+    private static final Logger LOG = LoggerFactory.getLogger(RemoteInfinispanTestSupport.class);
     public static final String DEFAULT_CACHE_NAME = "session_test_cache";
     public RemoteCache<String, SessionData> _cache;
     private String _name;
     public static RemoteCacheManager _manager;
+    private static final Logger INFINISPAN_LOG =
+        LoggerFactory.getLogger("org.eclipse.jetty.server.session.remote.infinispanLogs");
+
+    static GenericContainer infinispan;
 
     static
     {
         try
         {
-            String host = System.getProperty("hotrod.host", "127.0.0.1");
-
+            Testcontainers.exposeHostPorts(11222);
+            long start = System.currentTimeMillis();
+            String infinispanVersion = System.getProperty("infinispan.docker.image.version", "9.4.8.Final");
+            infinispan =
+                new GenericContainer(System.getProperty("infinispan.docker.image.name", "jboss/infinispan-server") +
+                                         ":" + infinispanVersion)
+                    .withEnv("APP_USER","theuser")
+                    .withEnv("APP_PASS","foobar")
+                    .withEnv("MGMT_USER", "admin")
+                    .withEnv("MGMT_PASS", "admin")
+                    .waitingFor(new LogMessageWaitStrategy()
+                                    .withRegEx(".*Infinispan Server.*started in.*\\s"))
+                    .withExposedPorts(4712,4713,8088,8089,8443,9990,9993,11211,11222,11223,11224)
+                    .withCopyFileToContainer(MountableFile.forClasspathResource("remote-session-test.xml"),
+                                              "/opt/jboss/infinispan-server/standalone/configuration/remote-session-test.xml")
```

Comment:
So if by default infinispan starts with standalone.xml file as the config, we don't need to use my remote-session-test.xml, correct?

Reply:
yes correct. I'm happy to remove this :)